src/main/io/osd.c

```diff
@@ -724,7 +725,9 @@ static bool osdDrawSingleElement(uint8_t item)
         if (!ARMING_FLAG(ARMED)) {
             tfp_sprintf(buff, "DISARMED");
         } else {
-            tfp_sprintf(buff, "        ");
+            if (!lastArmState) {  // previously disarmed - blank out the message one time
```

Comment:
The problem with only drawing it once is that any elements “behind” it will begin the consume it.   In the original issue the artificial horizon would eat it.  Also all other elements (even static) redraw every iteration.

I think there’s an optimization opportunity in general but it would require keeping a “previous state” for each element and not drawing them if they’re unchanged - but not a small change.

I also think the element layering order should be made consistent between the firmware and the configurator because they’re not now.  For example the firmware displays the “DISARMED” above the artificial horizon, but it’s reversed in the configurator.